### PR TITLE
375 Switch to Google OAUth2

### LINKED ISSUE #375: Login with Google account doesn't work anymore
Tried it, ended up at https://support.google.com/accounts/answer/6206245?p=openid&rd=1 which has title *OpenID 2.0 for Google Accounts has gone away*.   The page points developers at https://developers.google.com/identity/protocols/OpenID2Migration#shutdown-timetable for information on converting to the new system.

### PR DESCRIPTION
**Pull request by dpoirier**
Google OpenID has been turned off. Clean out old
associations using Google OpenID, and set up
Google OAuth2.

Fixes #375

### CODE REVIEW

File: pycon/migrations/0002_remove_old_google_openid_auths.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from django.db import migrations
+
+
+def no_op(apps, schema_editor):
+    pass
+
+
+def remove_old_google_openid_auths(apps, schema_editor):
+    UserSocialAuth = apps.get_model('social_auth', 'UserSocialAuth')
+    db_alias = schema_editor.connection.alias
```
Comment (rebecca-caktus) [inline pycon/migrations/0002_remove_old_google_openid_auths.py]: As far as I know, we only have one database. Is this line needed?

File: pycon/settings/local.py-example [hunk 2]
```diff
@@ -18,3 +18,10 @@
 # Uncomment to use Django Debug Toolbar:
 #INSTALLED_APPS.append('debug_toolbar')
 #MIDDLEWARE_CLASSES.append('debug_toolbar.middleware.DebugToolbarMiddleware')
+
+# To use Google OAuth2 locally, see the comments in server.py about how to get a client
+# ID and secret, then uncomment these lines and either set the client creds in your
+# environment or edit them in here.
+# AUTHENTICATION_BACKENDS[1:1] = ["social_auth.backends.google.GoogleOAuth2Backend",]
```
Comment (rebecca-caktus) [inline pycon/settings/local.py-example]: I had to try this out on the command line to see what it does. Perhaps `insert` would be more canonical?

**Comment by rebecca-caktus** [comment]
Apparently `django-social-auth` has been deprecated and should be replaced with `python-social-auth`:

https://github.com/omab/django-social-auth
https://github.com/omab/python-social-auth

File: pycon/settings/base.py [hunk 3]
```diff
@@ -253,7 +253,8 @@ def env_or_default(NAME, default):
     "symposion.teams.backends.TeamPermissionsBackend",
 
     # Social Auth Backends
-    "social_auth.backends.google.GoogleBackend",
+    # "social_auth.backends.google.GoogleBackend",  # Google OpenID support - obsolete
+    # "social_auth.backends.google.GoogleOAuth2Backend",  # We insert this one in server.py
```
Comment (rebecca-caktus) [inline pycon/settings/base.py]: Why not insert this one in the base setting as well? Yahoo & OpenID are there.

File: pycon/settings/server.py [hunk 4]
```diff
@@ -99,3 +99,31 @@
...
+# (4) Wait until the page says the project has been created
+# (5) The project will open in the dev console
+# (6) On the left, select "APIs & auth"/"Credentials"
+# (7) Under OAuth, click "Create new Client ID" (don't worry, it's OAuth2)
+# (8) Select application type "Web application"
+# (9) Enter a Product Name on the consent Screen page and click Save
+# (10) In the Create Client ID popup, select
+#     Application type:  Web application
+#     Authorized JS origins: Your site base URL (e.g. https://staging-pycon.python.org, not
+#       https://staging-pycon.python.org/2016/;  https://example.com or
```
Comment (rebecca-caktus) [inline pycon/settings/server.py]: That's a lot of examples.

File: pycon/settings/base.py [hunk 5]
```diff
@@ -253,7 +253,8 @@ def env_or_default(NAME, default):
     "symposion.teams.backends.TeamPermissionsBackend",
 
     # Social Auth Backends
-    "social_auth.backends.google.GoogleBackend",
+    # "social_auth.backends.google.GoogleBackend",  # Google OpenID support - obsolete
```
Comment (rebecca-caktus) [inline pycon/settings/base.py]: Any reason to leave this line in at all?

File: pycon/settings/base.py [hunk 6]
```diff
@@ -359,3 +361,7 @@ def env_or_default(NAME, default):
 # This can be removed in 1.8, the warning has been
 # removed.
 TEST_RUNNER = 'django.test.runner.DiscoverRunner'
+
+# Need to switch from the now-default serializer or OAuth2 breaks trying
+# to serialize a datetime
+SESSION_SERIALIZER='django.contrib.sessions.serializers.PickleSerializer'
```
Comment (rebecca-caktus) [inline pycon/settings/base.py]: Will this require clearing sessions or anything of the such?

File: pycon/settings/base.py [hunk 7]
```diff
@@ -279,6 +280,7 @@ def env_or_default(NAME, default):
 ACCOUNT_USER_DISPLAY = lambda user: user.get_full_name()
 LOGIN_ERROR_URL = reverse_lazy("account_login")
 
+
```
Comment (rebecca-caktus) [inline pycon/settings/base.py]: Extra space probably not needed.

File: pycon/migrations/0002_remove_old_google_openid_auths.py (hunk 1 above)
Comment (dpoirier) [inline pycon/migrations/0002_remove_old_google_openid_auths.py]: On Thu, Jul 23, 2015 at 12:57 PM, Rebecca <notifications@github.com> wrote:

> > +    db_alias = schema_editor.connection.alias
>
> As far as I know, we only have one database. Is this line needed?
>

​No, just me being pedantic in Django.

**Comment by dpoirier** [comment]
Opened #419 to maybe upgrade to python-social-auth some day.

File: pycon/settings/base.py (hunk 6 above)
Comment (dpoirier) [inline pycon/settings/base.py]: I haven't noticed any issues running locally. And the SESSION_SERIALIZER docs don't appear to have any warnings about changing it.

**Comment by rebecca-caktus** [comment]
:key:

**Pull request merged**